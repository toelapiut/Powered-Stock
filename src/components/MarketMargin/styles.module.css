
.bear, .bull{
    @apply rounded
}
.bull{
    @apply bg-green;
}
.bear{
    @apply bg-red;
}
.container{
    @apply px-2 pt-1
}

.margin{
    @apply font-bold text-xs leading-4
}