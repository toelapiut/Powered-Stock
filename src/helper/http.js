import axios from 'axios';

export const quandl = axios.create({
  baseURL: process.env.process.env.REACT_APP_BASE_URL,
  headers: {
    'Content-Type': 'application/json',
    'Accept': 'application/json',
    'Allow': 'GET, PUT, HEAD, OPTIONS'
  }
});

quandl.interceptors.request.use((config) => {
  config.params = config.params || {};
  config.params['key'] = process.env.REACT_APP_API_KEY;
  return config;
});
