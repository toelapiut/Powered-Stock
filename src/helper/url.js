export const url = {
  markets: 'datatables'
};

export const types = {
  ZACKS_CP: 'ZACKS/CP',
  ZACKS_MT: 'ZACKS/MT'
};