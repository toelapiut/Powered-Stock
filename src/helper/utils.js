import humps from 'humps';
import checkPropTypes from 'check-prop-types';
import {timeFormat} from 'd3-time-format';
import {bisector} from 'd3-array';
import moment from 'moment';

export const transpose = (rows, cols) => {
  /*
   * check if the row and col are empty and throw an error to
   * inform of the necessary requirements
   */
  if (typeof rows === 'undefined' || typeof cols === 'undefined') {
    throw new Error('row and col can not be undefined');
  }

  /*
   * throw error if the array isn't an array or col isn't an array
   */
  if (!Array.isArray(rows)) throw new Error('row must be  array');
  if (!Array.isArray(cols)) throw new Error('col must be  array');

  let dataset = [];

  for (let row of rows) {
    /*
     * check the data type before iterating through the matrix
     */
    if (!Array.isArray(row)) throw new Error('row must be a matrix row: [[a, b]]');
    let obj = {};
    row.forEach((item, index) => {
      obj[cols[index]['name']] = item;
    });
    dataset.push(obj);
  }

  return dataset;
};


export const transposeStock = (rows) => {
  let data = [];
  for (let row of rows) {
    let obj = {};
    const [date, close] = row;
    obj['date'] = date;
    obj['close'] = close;
    data.push(obj);
  }
  return data;
};

export const toCamelCase = (item) => {
  if (typeof item === 'undefined') {
    return item;
  }
  return humps.camelizeKeys(item);
};

export const checkPositivity = (num) => {
  if (typeof num === 'undefined') throw new Error('argument must not be empty');
  if (typeof num !== 'number') throw new Error('argument is not a number');
  let check = Math.sign(Number(num));
  return check !== -1;
};

export const formatDate = timeFormat('%b %d, \'%y');

export const getDate = (d) => new Date(d.date);

export const getStockValue = (d) => d.close;

export const bisectDate = bisector(d => new Date(d.date)).left;

export const getWindowDimensions = () => {
  const {innerWidth: width, innerHeight: height} = window;
  return {
    width,
    height
  };
};

export const dateFormat = (date, format = 'DD MMM, YYYY') => {
  return moment(date).format(format);
};

export const findTestByAttr = (component, attr) => {
  return component.find(`[data-test='${attr}']`);
};

export const checkProps = (component, expectedProps) => {
  checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
};

