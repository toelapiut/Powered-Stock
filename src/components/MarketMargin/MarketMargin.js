import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.module.css';
import {checkPositivity} from '../../helper/utils';

export const MarketMargin = ({margin}) => {
  let className = checkPositivity(margin) ? styles.bull : styles.bear;
  return (
    <div className={`${className} ${styles.container}`}>
      <p className={styles.margin}>{margin}</p>
    </div>
  );
};

MarketMargin.propTypes = {
  margin: PropTypes.number.isRequired
};