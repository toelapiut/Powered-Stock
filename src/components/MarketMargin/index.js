import {MarketMargin} from './MarketMargin';

export default MarketMargin;