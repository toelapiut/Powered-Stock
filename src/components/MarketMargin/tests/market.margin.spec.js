import React from 'react';
import MarketMargin from '../';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {findTestByAttr, checkProps} from '../../../helper/utils';


const setUp = (props = {}) => {
  return shallow(<MarketMargin {...props}/>);
};

let props = {
  margin:-123
};

describe('MarketMargin Component', () => {

  describe('Market Margin renders correctly', () => {
    let component;
    beforeAll(() => {
      component = setUp(props);
    });

    test('should render snapshot correctly', () => {
      expect(toJson(component)).toMatchSnapshot();
    });

    it('should render without error', () => {
      const wrapper = findTestByAttr(component, 'market-margin-container');
      expect(wrapper.length).toBe(1);
    });

  });

  describe('Check property types', () => {
    it('should not throw a warning', () => {
      const propErr = checkProps(MarketMargin, props);
      expect(propErr).toBeUndefined();
    });
  });
});