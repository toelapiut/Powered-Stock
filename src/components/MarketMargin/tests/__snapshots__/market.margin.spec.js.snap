// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketMargin Component Market Margin renders correctly should render snapshot correctly 1`] = `
<div
  className="bear container"
  data-test="market-margin-container"
>
  <p
    className="margin"
  >
    -123
  </p>
</div>
`;
